pkg/agent/externalnode/external_node_controller_linux_test.go

```diff
@@ -0,0 +1,1155 @@
+				mockIfaceStore.EXPECT().DeleteInterface(tt.preIf).Times(1)
+			} else if tt.interfaceAction == "add" {
+				mockIfaceStore.EXPECT().GetInterfaceByName(tt.curIf.InterfaceName).Return(tt.curIf, true)
+				portData := &ovsconfig.OVSPortData{
+					ExternalIDs: map[string]string{
+						ovsExternalIDUplinkName:      tt.curIf.InterfaceName,
+						ovsExternalIDUplinkPort:      tt.curIf.UplinkPort.PortUUID,
+						ovsExternalIDEntityName:      tt.curIf.EntityName,
+						ovsExternalIDEntityNamespace: tt.curEN.Namespace,
+						ovsExternalIDIPs:             strings.Join(curIPsStr, ipsSplitter),
+					},
+				}
+				mockOVSBridgeClient.EXPECT().GetPortData(tt.curIf.PortUUID, tt.curIf.InterfaceName).Return(portData, nil).Times(1)
+				expectedInfo := map[string]interface{}{
+					"uplink-name":      tt.curIf.InterfaceName,
+					"entity-name":      tt.curEN.Name,
+					"antrea-type":      "host",
+					"entity-namespace": tt.curEN.Namespace,
+					"ip-address":       strings.Join(curIPsStr, ipsSplitter),
+					"uplink-port":      tt.curIf.UplinkPort.PortUUID,
+				}
+				mockOVSBridgeClient.EXPECT().SetPortExternalIDs(tt.curIf.InterfaceName, expectedInfo).Times(1)
+				iface := &interfacestore.InterfaceConfig{
+					InterfaceName: tt.curIf.InterfaceName,
+					Type:          interfacestore.ExternalEntityInterface,
+					OVSPortConfig: tt.curIf.OVSPortConfig,
+					EntityInterfaceConfig: &interfacestore.EntityInterfaceConfig{
+						EntityName:      tt.curEN.Name,
+						EntityNamespace: tt.curIf.EntityNamespace,
+						UplinkPort:      tt.curIf.UplinkPort,
+					},
+					IPs: curIPs,
+				}
+				t.Logf("%+v", iface.EntityInterfaceConfig)
+				mockIfaceStore.EXPECT().AddInterface(iface).Times(1)
+				mockIfaceStore.EXPECT().GetInterfaceByName(tt.preIf.InterfaceName).Return(&if1, true).Times(1)
+				mockIfaceStore.EXPECT().GetInterfacesByType(interfacestore.ExternalEntityInterface).Return(
+					[]*interfacestore.InterfaceConfig{tt.curIf},
+				).Times(1)
+			}
+			err := c.updateExternalNode(tt.preEN, tt.curEN)
+			assert.NoError(t, err)
+			assert.Equal(t, tt.syncedExternalNode, c.syncedExternalNode)
+		})
+	}
+}
+
+func TestAddInterface(t *testing.T) {
+	_, cidr, _ := net.ParseCIDR("172.16.1.0/24")
+	for _, tt := range []struct {
+		name                    string
+		iface                   *interfacestore.InterfaceConfig
+		eeNamespace             string
+		eeName                  string
+		IPs                     []string
+		preIPs                  []string
+		portData                *ovsconfig.OVSPortData
+		uplinkName              string
+		preEEName               string
+		uplinkPortUUID          string
+		ifaceExist              bool
+		getInterfaceConfigParam mockGetInterfaceConfigParam
+		ifAddInterface          bool
+	}{
+		{
+			name:           "add interface successfully",
+			iface:          &if1,
+			ifaceExist:     true,
+			eeNamespace:    "eeNamespace",
+			eeName:         "eeName",
```

Comment:
Do you need these two fields if all values are the same in different cases?

Reply:
updated. Put them to the vars